src/Build/BuildCheck/Infrastructure/CustomConfigurationData.cs

```diff
@@ -55,10 +65,55 @@ public override bool Equals(object? obj)
             return false;
         }
 
-        return Equals((CustomConfigurationData)obj);
+        var customConfigObj = (CustomConfigurationData) obj;
+
+        if (customConfigObj.RuleId != RuleId)
+        {
+            return false;
+        }
+
+        // validate keys and values
+        if (customConfigObj.ConfigurationData != null && ConfigurationData != null && ConfigurationData.Count == customConfigObj.ConfigurationData.Count)
+        {
+            foreach (var keyVal in customConfigObj.ConfigurationData)
+            {
+                if (!ConfigurationData.TryGetValue(keyVal.Key, out var value) || value != keyVal.Value)
+                {
+                    return false;
+                }
+            }
+        }
+        else if (customConfigObj.ConfigurationData == null && ConfigurationData == null)
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+
+        return true;
     }
 
-    protected bool Equals(CustomConfigurationData other) => Equals(ConfigurationData, other.ConfigurationData);
+    public override int GetHashCode()
+    {
+        if (!NotNull(this))
+        {
+            return 0;
+        }
+
+        var hashCode = RuleId.GetHashCode();
+        if (ConfigurationData != null)
+        {
+            foreach (var keyVal in ConfigurationData)
+            {
+                unchecked
+                {
+                    hashCode = hashCode + keyVal.Key.GetHashCode() + keyVal.Value.GetHashCode();
```

Comment:
This is quite suboptimal implementation. If we want to override it, we should keep good hash distribution.

Have a look on `HashCode.Combine`

Reply:
Thank you! I'll check the It was discussed with @ladipro during the PR review, and he suggested to throw not implemented solution, which I'm more and more leaning toward to right now. GetHashCode method will be used only in collections based on hash table, having also that the class is public, shall we limit the usage of this at the moment (by suggested approach? ) what do you think?

Reply:
Per discussion, for now the code is removed, if there will be a need to actually use CustomConfiguration data in a collections based on hashTable we will consider readding it. 
Thank you